pkg/spec/resources.go

```diff
@@ -417,7 +417,38 @@ func (app *App) createServices() ([]runtime.Object, error) {
 				}
 				svcs = append(svcs, endpointIngress)
 			}
+
+			if port.RouteEndpoint != "" {
+
+				endpointRoute := &os_route_v1.Route{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   svc.Name,
+						Labels: app.Labels,
+					},
+					Spec: os_route_v1.RouteSpec{
+						To: os_route_v1.RouteTargetReference{
+							Kind: "Service",
+							Name: svc.Name,
+						},
+					},
+				}
+
+				switch port.RouteEndpoint {
+				case "true":
+					// OpenShift will take care of route
+					svcs = append(svcs, endpointRoute)
+				case "false":
+					// Do not create route
+					log.Debug("Don't create route")
+				default:
+					// Route URL is given by user
+					endpointRoute.Spec.Host = port.RouteEndpoint
+					svcs = append(svcs, endpointRoute)
+
```

Comment:
i believe you can shorten this to an if statement instead.. ex:

```go
if port.RouteEndpoint {
 svcs = append(svcs, endpointRoute)
} else {
 endpointRoute.Spec.Host = port.RouteEndpoint
 svcs = append(svcs, endpointRoute)
}
```

Reply:
@cdrage , since port.RouteEndpoint is string, I can't do that